tests/callbacks/test_device_stats_monitor.py

```diff
@@ -54,6 +62,64 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
     trainer.fit(model)
 
 
+@RunIf(psutil=True)
+@pytest.mark.parametrize("cpu_stats", (None, True, False))
+def test_device_stats_cpu(tmpdir, cpu_stats):
+    """Test CPU stats are logged when no accelerator is used."""
+    model = BoringModel()
+    CPU_METRIC_KEYS = [
+        _CPU_VM_PERCENT,
+        _CPU_SWAP_PERCENT,
+        _CPU_PERCENT,
+    ]
+
+    class DebugLogger(CSVLogger):
+        def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+            enabled = cpu_stats or cpu_stats is None
+            for f in CPU_METRIC_KEYS:
+                has_cpu_metrics = any(f in h for h in metrics)
+                assert has_cpu_metrics if enabled else not has_cpu_metrics
+
+    device_stats = DeviceStatsMonitor(cpu_stats=cpu_stats)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_train_batches=2,
+        limit_val_batches=0,
+        log_every_n_steps=1,
+        callbacks=device_stats,
+        logger=DebugLogger(tmpdir),
+        enable_checkpointing=False,
+        enable_progress_bar=False,
+        accelerator="cpu",
+    )
+    trainer.fit(model)
+
+
+@RunIf(psutil=True)
+@mock.patch("pytorch_lightning.accelerators.cpu.get_cpu_process_metrics")
+@mock.patch("pytorch_lightning.callbacks.device_stats_monitor.get_cpu_process_metrics")
+def test_device_stats_cpu_queried_once(cpu_metrics_device_stats_mock, cpu_metrics_cpu_accelerator_mock, tmpdir):
```

Comment:
I think this one and the above one can be merged.

Reply:
Done